docs/reference/controllers.md

```diff
@@ -197,6 +197,40 @@ class EffectsController extends Controller {
 }
 ```
 
+`dispatch` accepts additional options as the second parameter as follows:
+
+option       | default            | notes
+-------------|--------------------|----------------------------------------------------------------------------------------------
+`detail`     | `{}` empty object  | See [CustomEvent.detail](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/detail)
+`target`     | `this.element`     | See [Event.target](https://developer.mozilla.org/en-US/docs/Web/API/Event/target)
+`prefix`     | `this.identifier`  | If prefix is falsey (e.g. `null`), only the `eventName` will be used
+`bubbles`    | `true`             | See [Event.bubbles](https://developer.mozilla.org/en-US/docs/Web/API/Event/bubbles)
+`cancelable` | `true`             | see [Event.cancelable](https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelable)
+
+`dispatch` will return the generated [`CustomEvent`](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent), you can use this to provide a way for the event to be cancelled by any other listeners as follows:
+
+```js
+class ClipboardController extends Controller {
+  static targets = [ "source" ]
+
+  copy() {
+    const event = this.dispatch("copy", { cancelable: true })
+    if (event.defaultPrevented) return;
```

Comment:
I just looked at the code snippets in the docs, it seems like we are not using semi-colons.

```suggestion
    if (event.defaultPrevented) return

```